src/twisted/trial/test/test_tests.py

```diff
@@ -965,6 +965,28 @@ def test_multipleErrorsReported(self):
         self.assertEqual(error2.getErrorMessage(), "foo")
 
 
+    def test_cleanupsAreReentrant(self):
+        @self.test.addCleanup
+        def reenter():
+            self.test.append("foo")
+            self.test.addCleanup(self.test.append, "bar")
+
+        self.test.run(self.result)
+        self.assertEqual(["setUp", "runTest", "foo", "bar", "tearDown"], self.test.log)
+
+    def test_cleanupsIgnoreKeyboardInterrupt(self):
```

Comment:
```suggestion
    def test_cleanupsHandleKeyboardInterrupt(self):
```

Reply:
I think it would be better to split off the `KeyboardInterrupt` changes into a new PR, since this is a bug fix which should be documented in the release notes.

Reply:
Agreed, shall I reset to https://github.com/twisted/twisted/pull/1388/commits/450e0e118e5539ebf46127d5b98ece58e04aa066 ?